src/harness/sas_testcase.py

```diff
@@ -282,6 +282,49 @@ def assertChannelsContainFrequencyRange(self, channels, frequency_range):
     self.assertEqual(channels[0]['frequencyRange']['highFrequency'],
                      frequency_range['highFrequency'])
 
+  def assertChannelsOverlapFrequencyRange(self, channels, frequency_range,
+                                          constrain_low=False, constrain_high=False):
+    """Checks if the frequency range of all channels overlap
+       the given frequency range.
+
+    Args:
+      channels: A list of dictionaries containing frequencyRange,
+        which is a dictionary containing lowFrequency and highFrequency.
+      frequency_range: A dictionary containing lowFrequency and highFrequency.
+      constrain_low: A Boolean flag indicating if the lower frequency edge of the
+        channel with the lowest frequency segment need to be constrained to be 
+        equal to the lower frequency edge of the frequency range.
+      constrain_high: A Boolean flag indicating if the upper frequency edge of the
+        channel with the highest frequency segment need to be constrained to be 
+        equal to the upper frequency edge of the frequency range.
+    """
+    channels.sort(
+        key=
+        lambda ch: (ch['frequencyRange']['lowFrequency'],
+                    ch['frequencyRange']['highFrequency']),
+        reverse=False)
+    for index, channel in enumerate(channels):
+      if index == 0:
+        if constrain_low:
+          self.assertEqual(channel['frequencyRange']['lowFrequency'],
+                           frequency_range['lowFrequency'])
+        else:
+          self.assertGreaterEqual(channel['frequencyRange']['lowFrequency'],
+                                  frequency_range['lowFrequency'])
+      else:
+        self.assertGreaterEqual(
```

Comment:
Why are you reversing the direction of the comparison? Is there a bug in `assertChannelsContainFrequencyRange()`?

Reply:
Thanks for pointing out, it was an leftover mistake. Fixed.